applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -1306,12 +1317,58 @@ int unicast_client_stop(void)
 	return 0;
 }
 
+static bool ts_valid(enum timestamp_state ts_state)
+{
+	return (ts_state == TS_VALID);
+}
+
+#define SDU_INTERVAL_US (10000u)
+/**
+ * @brief
+ *
+ * @param tx_info
+ * @param ts_state
+ * @return uint32_t
+ */
+uint32_t timestamp_select(struct bt_iso_tx_info const *const tx_info)
+{
+
+	if (!ts_valid(ts_state[AUDIO_CH_L]) && !ts_valid(ts_state[AUDIO_CH_R])) {
+		LOG_WRN("No valid timestamp found");
+		return 0;
+	}
+
+	if (ts_valid(ts_state[AUDIO_CH_L]) && !ts_valid(ts_state[AUDIO_CH_R])) {
+		return tx_info[AUDIO_CH_L].ts;
+	}
+
+	if (!ts_valid(ts_state[AUDIO_CH_L]) && ts_valid(ts_state[AUDIO_CH_R])) {
+		return tx_info[AUDIO_CH_R].ts;
+	}
+
+	/* From this point, we know that we have valid ts_state for both channels.
+	 * We need to check for wrapping. If both values have wrapped, this is fine.
+	 * However, if only one has wrapped, we must select the smaller value.
+	 */
+
+	uint32_t diff = abs((int)(tx_info[AUDIO_CH_L].ts - tx_info[AUDIO_CH_R].ts));
+
+	if (diff > (UINT32_MAX / 2)) {
```

Comment:
To my understanding, they may not have wrapped if the difference is larger than `UINT32_MAX / 2 `but smaller than `UINT32_MAX`. Is there an underlying assumption that allows this if to check for wrapping? Or am I misunderstanding something?

Reply:
Discuss

Reply:
We have talked, and have found an alternative way to handle this. Fix will be pushed today.